src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -414,62 +415,90 @@ internal static bool IsModuleMatchingModuleSpec(PSModuleInfo moduleInfo, ModuleS
                 {
-                    var dataFileSetting =
-                        PsUtils.GetModuleManifestProperties(
-                            manifestPath,
-                            PsUtils.ManifestGuidPropertyName);
+                    Guid guidID;
+                    if (LanguagePrimitives.TryConvertTo(guidValue, out guidID))
+                    {
+                        return guidID;
+                    }
+                }
+            }
+            catch (PSInvalidOperationException) { }
 
-                    var guidValue = dataFileSetting["GUID"];
-                    if (guidValue != null)
+            return new Guid();
+        }
+
+        internal static ExperimentalFeature[] GetExperimentalFeature(string manifestPath)
+        {
+            try
+            {
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestPrivateDataPropertyName);
+
+                object privateData = dataFileSetting["PrivateData"];
+                if (privateData is Hashtable hashData && hashData["PSData"] is Hashtable psData)
+                {
+                    object expFeatureValue = psData["ExperimentalFeatures"];
+                    if (expFeatureValue != null &&
+                        LanguagePrimitives.TryConvertTo(expFeatureValue, out Hashtable[] features) &&
+                        features.Length > 0)
                     {
-                        Guid guidID;
-                        if (LanguagePrimitives.TryConvertTo(guidValue, out guidID))
+                        string moduleName = ModuleIntrinsics.GetModuleName(manifestPath);
+                        var expFeatureList = new List<ExperimentalFeature>();
+                        foreach (Hashtable feature in features)
                         {
-                            return guidID;
+                            string featureName = feature["Name"] as string;
+                            if (String.IsNullOrEmpty(featureName)) { continue; }
+
+                            if (ExperimentalFeature.IsModuleFeatureName(featureName, moduleName))
+                            {
+                                string featureDescription = feature["Description"] as string;
+                                expFeatureList.Add(new ExperimentalFeature(featureName, featureDescription, manifestPath,
+                                                                           ExperimentalFeature.HasEnabled(featureName)));
+                            }
                         }
+                        return expFeatureList.ToArray();
                     }
                 }
-                catch (PSInvalidOperationException)
-                {
-                }
             }
+            catch (PSInvalidOperationException) { }
 
-            return new Guid();
+            return Utils.EmptyArray<ExperimentalFeature>();
```

Comment:
I know we already have a utility method for this and you're just calling it, but how is it different from `Array.Empty<ExperimentalFeature>()`?

Reply:
A-ha! There is no difference. I just found that `Array.Empty<T>` became available since 4.6 ...
I will submit a separate PR to change all instance of `Utils.EmptyArray` to `Array.Empty`.

Reply:
Yeah, I understand. Update those is spending effort with little value (and there is rick of regression if a mistake is made during the update).
But there is one benefit -- when people see the code, they will know "OK, .NET has this cool method that I can use elsewhere". Currently, some people might not realize that they can do the same outside of PowerShell repository.